File: Unit2-01/mrCoxallStack.ts
/*
 * This program is a number guessing game
 * @author  Ava Venturino 
 * @version 1.0 
 * @since   2024-02-26
 */

class MrCoxallStack {
  private stackAsList: string[] = [];

  push(pushedString: string): void {
    // push the value onto the stack
    this.stackAsList.push(pushedString);
  }

  showStack(): string {
    // return the stack
    let stackValues: string = "";
	        
    for (const value of this.stackAsList) {
      stackValues += value + ",";
    }
    stackValues = stackValues.slice(0, -1); // Remove the trailing comma

    return stackValues;
  }
}

File: Unit2-01/MrCoxallStack.java
/*
 * This is a program prints out stuff.
 * @author  Ava Venturino 
 * @version 1.0 
 * @since   2024-02-26
 */

import java.util.ArrayList;
import java.util.List;

public class MrCoxallStack {
    private List<String> stackAsList;

    public MrCoxallStack() {
        this.stackAsList = new ArrayList<>();
    }

    public void push(String pushedString) {
        // push the value onto the stack
        this.stackAsList.add(pushedString);
    }

    public String showStack() {
        // return the stack
        StringBuilder stackValues = new StringBuilder();

        for (String value : this.stackAsList) {
            stackValues.append(value).append(",");
        }
		        
        if (stackValues.length() > 0) {
            // Remove the trailing comma
            stackValues.deleteCharAt(stackValues.length() - 1);
        }

        return stackValues.toString();
    }
}
